### PR TITLE
Improve handling build id in view page

### PR DESCRIPTION
**Pull request by pditommaso**
This PR allows the accessing the build page given the container id. For example having the container `HOST/wt/d67234213cdb/wave/build/dev:a97082cddd28bacf`, the id `a97082cddd28bacf` can be used to access the URI `https://HOST/view/builds/a97082cddd28bacf_1`

### CODE REVIEW

**Comment by pditommaso** [comment]
@munishchouhan I need some help with this Surreal query 

https://github.com/seqeralabs/wave/blob/314dd4c4694a290113d15b0d96c19ab02872f19a/src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy#L151-L151

I'd like to have the equivalent of `buildId like '${something}_%' in SQL, but not sure there's a `like` operator in Surreal

**Comment by munishchouhan** [comment]
we can use `string::contains(string, string) -> bool`
like:
**seqera/wave> select * from wave_build where string::contains( buildId,'b6266b6b8aead019_')**
```
[[{ buildId: 'b6266b6b8aead019_1', condaFile: 'channels:
- conda-forge
- bioconda
dependencies:
- bwa
', digest: sha256:ab5e16ac0f300c1054bb9ee382b866314680fe118465b956624121fb264e75a5, dockerFile: 'FROM mambaorg/micromamba:1.5.8-lunar
COPY --chown=$MAMBA_USER:$MAMBA_USER conda.yml /tmp/conda.yml
RUN micromamba install -y -n base -f /tmp/conda.yml \\
    && micromamba install -y -n base conda-forge::procps-ng \\
    && micromamba clean -a -y
USER root
ENV PATH="$MAMBA_ROOT_PREFIX/bin:$PATH"
', duration: 59.042196f, exitStatus: 0, format: 'DOCKER', id: wave_build:d9ec62mqy0nimjjgy6i1, offsetId: '+02:00', platform: 'linux/arm64', requestIp: '127.0.0.1', scanId: '61bec051aded', startTime: '2024-09-18T11:04:05.827099Z', targetImage: 'docker.io/hrma017/dev:bwa--b6266b6b8aead019' }]]
```

**Comment by munishchouhan** [comment]
https://surrealdb.com/docs/surrealql/functions/database/string

File: src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy [hunk 1]
```diff
@@ -146,6 +146,16 @@ class SurrealPersistenceService implements PersistenceService {
         return result
     }
 
+    @Override
+    WaveBuildRecord latestBuild(String containerId) {
+        final query = "select * from wave_build where buildId ~ '${containerId}${BuildRequest.SEP}' order by startTime desc limit 1"
```
Comment (munishchouhan) [inline src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy:151]:
```suggestion
        final query = "select * from wave_build where string::contains(buildId,'${containerId}${BuildRequest.SEP}') order by startTime desc limit 1"
```

**Comment by pditommaso** [comment]
@munishchouhan not sure `string::contains` is working as expected

**Comment by munishchouhan** [comment]
> @munishchouhan not sure `string::contains` is working as expected

ok i will check again

**Comment by pditommaso** [comment]
I've assumed that's the query because the build failed, not 100% sure tho

**Comment by munishchouhan** [comment]
I tested a build locally using this PR and it was successful

Screenshot 2024-09-18 at 18 27 17

**Comment by pditommaso** [comment]
Yeah, but the build is failing 😆

**Comment by munishchouhan** [comment]
> Yeah, but the build is failing 😆

ok you mean gradle build

**Comment by pditommaso** [comment]
@munishchouhan any chance to solve failing tests this PR ?

**Comment by munishchouhan** [comment]
> @munishchouhan any chance to solve failing tests this PR ?

I am off till monday. I can do it on tuesday

**Pull request merged**